## Final Project Submission

Please fill out:


- Group 2


Student name: 


***Alvin	Kimathi***

***Lynn	Oloo***

***Hawkins	Murithi***

***Firdosa	Mohamed***


* Student pace: FULL TIME/HYBRID

* Scheduled project review date/time: 2nd MAY 2024

* Instructor name: MARYANN MWIKALI

* Blog post URL:


### DATA UNDERSTANDING

***Importing libraries***

In [1]:
import pandas as pd
import matplotlib as plt

***Reading the Kc_house_data.csv Dataset***

In [2]:
df = pd.read_csv("data/kc_house_data.csv", index_col = 0)
df.head()

date     price  bedrooms  bathrooms  sqft_living  sqft_lot  \
id                                                                             
7129300520  10/13/2014  221900.0         3       1.00         1180      5650   
6414100192   12/9/2014  538000.0         3       2.25         2570      7242   
5631500400   2/25/2015  180000.0         2       1.00          770     10000   
2487200875   12/9/2014  604000.0         4       3.00         1960      5000   
1954400510   2/18/2015  510000.0         3       2.00         1680      8080   

            floors waterfront  view  condition          grade  sqft_above  \
id                                                                          
7129300520     1.0        NaN  NONE    Average      7 Average        1180   
6414100192     2.0         NO  NONE    Average      7 Average        2170   
5631500400     1.0         NO  NONE    Average  6 Low Average         770   
2487200875     1.0         NO  NONE  Very Good      7 Average        1050   
1954400510     1.0         NO  NONE    Average         8 Good        1680   

           sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
id                                                                            
7129300520           0.0      1955           0.0    98178  47.5112 -122.257   
6414100192         400.0      1951        1991.0    98125  47.7210 -122.319   
5631500400           0.0      1933           NaN    98028  47.7379 -122.233   
2487200875         910.0      1965           0.0    98136  47.5208 -122.393   
1954400510           0.0      1987           0.0    98074  47.6168 -122.045   

            sqft_living15  sqft_lot15  
id                                     
7129300520           1340        5650  
6414100192           1690        7639  
5631500400           2720        8062  
2487200875           1360        5000  
1954400510           1800        7503

In [3]:
df.describe()

price      bedrooms     bathrooms   sqft_living      sqft_lot  \
count  2.159700e+04  21597.000000  21597.000000  21597.000000  2.159700e+04   
mean   5.402966e+05      3.373200      2.115826   2080.321850  1.509941e+04   
std    3.673681e+05      0.926299      0.768984    918.106125  4.141264e+04   
min    7.800000e+04      1.000000      0.500000    370.000000  5.200000e+02   
25%    3.220000e+05      3.000000      1.750000   1430.000000  5.040000e+03   
50%    4.500000e+05      3.000000      2.250000   1910.000000  7.618000e+03   
75%    6.450000e+05      4.000000      2.500000   2550.000000  1.068500e+04   
max    7.700000e+06     33.000000      8.000000  13540.000000  1.651359e+06   

             floors    sqft_above      yr_built  yr_renovated       zipcode  \
count  21597.000000  21597.000000  21597.000000  17755.000000  21597.000000   
mean       1.494096   1788.596842   1970.999676     83.636778  98077.951845   
std        0.539683    827.759761     29.375234    399.946414     53.513072   
min        1.000000    370.000000   1900.000000      0.000000  98001.000000   
25%        1.000000   1190.000000   1951.000000      0.000000  98033.000000   
50%        1.500000   1560.000000   1975.000000      0.000000  98065.000000   
75%        2.000000   2210.000000   1997.000000      0.000000  98118.000000   
max        3.500000   9410.000000   2015.000000   2015.000000  98199.000000   

                lat          long  sqft_living15     sqft_lot15  
count  21597.000000  21597.000000   21597.000000   21597.000000  
mean      47.560093   -122.213982    1986.620318   12758.283512  
std        0.138552      0.140724     685.230472   27274.441950  
min       47.155900   -122.519000     399.000000     651.000000  
25%       47.471100   -122.328000    1490.000000    5100.000000  
50%       47.571800   -122.231000    1840.000000    7620.000000  
75%       47.678000   -122.125000    2360.000000   10083.000000  
max       47.777600   -121.315000    6210.000000  871200.000000

In [4]:
df.shape

(21597, 20)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21597 entries, 7129300520 to 1523300157
Data columns (total 20 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   date           21597 non-null  object 
 1   price          21597 non-null  float64
 2   bedrooms       21597 non-null  int64  
 3   bathrooms      21597 non-null  float64
 4   sqft_living    21597 non-null  int64  
 5   sqft_lot       21597 non-null  int64  
 6   floors         21597 non-null  float64
 7   waterfront     19221 non-null  object 
 8   view           21534 non-null  object 
 9   condition      21597 non-null  object 
 10  grade          21597 non-null  object 
 11  sqft_above     21597 non-null  int64  
 12  sqft_basement  21597 non-null  object 
 13  yr_built       21597 non-null  int64  
 14  yr_renovated   17755 non-null  float64
 15  zipcode        21597 non-null  int64  
 16  lat            21597 non-null  float64
 17  long           21597 non-null  float

The dataset contains 21,597 entries and 20 columns. Some columns have missing values, such as "waterfront", "view", and "yr_renovated". The data types include integers, floats, and objects (likely representing categorical variables).

***Reading the column_names.md Dataset***

In [6]:
with open('data/column_names.md', 'r') as file:
    md_lines = file.readlines()

df_col = pd.DataFrame({'Text': md_lines})

df_col

Text
0   # Column Names and Descriptions for King Count...
1            * `id` - Unique identifier for a house\n
2                    * `date` - Date house was sold\n
3        * `price` - Sale price (prediction target)\n
4                 * `bedrooms` - Number of bedrooms\n
5               * `bathrooms` - Number of bathrooms\n
6   * `sqft_living` - Square footage of living spa...
7          * `sqft_lot` - Square footage of the lot\n
8   * `floors` - Number of floors (levels) in house\n
9   * `waterfront` - Whether the house is on a wat...
10    * Includes Duwamish, Elliott Bay, Puget Soun...
11            * `view` - Quality of view from house\n
12    * Includes views of Mt. Rainier, Olympics, C...
13  * `condition` - How good the overall condition...
14    * See the [King County Assessor Website](htt...
15  * `grade` - Overall grade of the house. Relate...
16    * See the [King County Assessor Website](htt...
17  * `sqft_above` - Square footage of house apart...
18  * `sqft_basement` - Square footage of the base...
19         * `yr_built` - Year when house was built\n
20  * `yr_renovated` - Year when house was renovat...
21  * `zipcode` - ZIP Code used by the United Stat...
22                    * `lat` - Latitude coordinate\n
23                  * `long` - Longitude coordinate\n
24  * `sqft_living15` - The square footage of inte...
25  * `sqft_lot15` - The square footage of the lan...

In [7]:
df_col.describe()

Text
count                                                  26
unique                                                 26
top     * `yr_renovated` - Year when house was renovat...
freq                                                    1

In [8]:
df_col.shape

(26, 1)

In [9]:
df_col.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26 entries, 0 to 25
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Text    26 non-null     object
dtypes: object(1)
memory usage: 336.0+ bytes


The provided dataset consists of a single column named "Text" with 26 entries. Each entry is of type object (likely representing strings or text data). There are no missing values in the dataset. It seems to be a small dataset containing textual data, possibly representing individual text samples, comments, or descriptions.

## Data cleaning

At this point check for the existing columns and understand what the entail.

In [10]:
df.columns

Index(['date', 'price', 'bedrooms', 'bathrooms', 'sqft_living', 'sqft_lot',
       'floors', 'waterfront', 'view', 'condition', 'grade', 'sqft_above',
       'sqft_basement', 'yr_built', 'yr_renovated', 'zipcode', 'lat', 'long',
       'sqft_living15', 'sqft_lot15'],
      dtype='object')

Choose the best-looking predictors to include in the model.

In [17]:
new_df = df[['price', 'bedrooms', 'sqft_living', 'sqft_lot', 'view', 'condition', 'grade','yr_built']]
new_df

price  bedrooms  sqft_living  sqft_lot  view  condition  \
id                                                                       
7129300520  221900.0         3         1180      5650  NONE    Average   
6414100192  538000.0         3         2570      7242  NONE    Average   
5631500400  180000.0         2          770     10000  NONE    Average   
2487200875  604000.0         4         1960      5000  NONE  Very Good   
1954400510  510000.0         3         1680      8080  NONE    Average   
...              ...       ...          ...       ...   ...        ...   
263000018   360000.0         3         1530      1131  NONE    Average   
6600060120  400000.0         4         2310      5813  NONE    Average   
1523300141  402101.0         2         1020      1350  NONE    Average   
291310100   400000.0         3         1600      2388  NONE    Average   
1523300157  325000.0         2         1020      1076  NONE    Average   

                    grade  yr_built  
id                                   
7129300520      7 Average      1955  
6414100192      7 Average      1951  
5631500400  6 Low Average      1933  
2487200875      7 Average      1965  
1954400510         8 Good      1987  
...                   ...       ...  
263000018          8 Good      2009  
6600060120         8 Good      2014  
1523300141      7 Average      2009  
291310100          8 Good      2004  
1523300157      7 Average      2008  

[21597 rows x 8 columns]

Use `.info()` to check for missing data in the columns

In [18]:
new_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21597 entries, 7129300520 to 1523300157
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   price        21597 non-null  float64
 1   bedrooms     21597 non-null  int64  
 2   sqft_living  21597 non-null  int64  
 3   sqft_lot     21597 non-null  int64  
 4   view         21534 non-null  object 
 5   condition    21597 non-null  object 
 6   grade        21597 non-null  object 
 7   yr_built     21597 non-null  int64  
dtypes: float64(1), int64(4), object(3)
memory usage: 1.5+ MB


we see that the column `view` has some missing data but since they are very few and will have little or no effect on our analysis we decide to use `dropna` to drop the rows.

In [22]:
new_df = new_df.dropna()

we again do `.info()` to see the output of our analysis.

In [23]:
new_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21534 entries, 7129300520 to 1523300157
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   price        21534 non-null  float64
 1   bedrooms     21534 non-null  int64  
 2   sqft_living  21534 non-null  int64  
 3   sqft_lot     21534 non-null  int64  
 4   view         21534 non-null  object 
 5   condition    21534 non-null  object 
 6   grade        21534 non-null  object 
 7   yr_built     21534 non-null  int64  
dtypes: float64(1), int64(4), object(3)
memory usage: 1.5+ MB


At this point we now see that all our columns have the same no of rows having non- null values hence we are ready to fodge a step forward.